src/renderers/dom/server/__tests__/ReactServerRendering-test.js

```diff
@@ -399,5 +399,38 @@ describe('ReactServerRendering', function() {
       );
       expect(markup.indexOf('hello, world') >= 0).toBe(true);
     });
+
+    it('allows setState in component rendered after another using renderToString',
+    function() {
+      var StaticComponent = React.createClass({
+        render: function() {
+          const staticContent = ReactServerRendering.renderToStaticMarkup(
+            <div>
+              <img src="foo-bar.jpg" />
+            </div>
+          );
+          // const staticContent = '<h1>Hello</h1>'
+          return <div dangerouslySetInnerHTML={{__html: staticContent}} />;
+        },
+      });
+
+      var Component = React.createClass({
+        componentWillMount: function() {
+          this.setState({text: 'hello, world'});
+        },
+        render: function() {
+          return <div>{this.state.text}</div>;
+        },
+      });
+      expect(
+        ReactServerRendering.renderToString.bind(
```

Comment:
Do you want me to change the other instances of this? I was just following the format the other tests in this file use (see [ReactServerRendering-test.js#L265-L268](https://github.com/Aweary/react/blob/a0eb88d6d0ef71258b78abc75c6f5cda9e44cfc6/src/renderers/dom/server/__tests__/ReactServerRendering-test.js#L265-L268))